pyeudiw/satosa/backend.py

```diff
@@ -215,14 +232,43 @@ def redirect_endpoint(self, context, *args):
             raise NoBoundEndpointError("request_uri not valid")
 
         # take the encrypted jwt, decrypt with my public key (one of the metadata) -> if not -> exception
-
+        
+        jwt = context.request["response"]
+        jwk = JWK(self.config["federation"]["federation_jwks"][0], "RSA")
+        
+        jweHelper = JWEHelper(jwk)
+        
+        decrypted_data = jweHelper.decrypt(jwt)
+        
         # get state and nonce, do lookup on the db -> if not -> exception
-
+        
+        nonce = decrypted_data.get("nonce", None) #TODO Fix this field handling
```

Comment:
the nonce is just in the vp token and not in the redirect response, I'll remove it